### PR TITLE
APPSEC_RUNTIME_ACTIVATION scenario

### PR DESCRIPTION
**Pull request by cataphract**
## Description

Add scenario where appsec is in the default inactive state and is enabled through remote config.

## Check list

- [x] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [x] CI is passing
- [x] There is at least one approval from code owners

### CODE REVIEW

File: scenarios/appsec/test_runtime_activation.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+from utils import BaseTestCase, context, coverage, interfaces, released, irrelevant
+
+# dd.rc.targets.key.id=TEST_KEY_ID
+# dd.rc.targets.key=1def0961206a759b09ccdf2e622be20edf6e27141070e7b164b7e16e96cf402c
+# private key: a78bd01afe0dc0baa6904e1b65448a6bbe160e07f7fc375c3bcb3ec08f008cc5
+
+
+@released(java="0.115.0", cpp="?", dotnet="2.16.0", php="?", python="?", ruby="?", nodejs="?", golang="?")
+@irrelevant(context.appsec_rules_file == "")
```
Comment (cbeauchesne) [inline scenarios/appsec/test_runtime_activation.py:13]: I'm not sure to understnad the purpose here ?

Comment (smola) [reply scenarios/appsec/test_runtime_activation.py:13]: If there are custom rules, AppSec's behavior is not updating them through remote config. So this scenario cannot be run properly with custom rules.

Comment (cbeauchesne) [reply scenarios/appsec/test_runtime_activation.py:13]: The point is that all scenario should run only relevant test classes. If a scenario uses a custom rules set, then it should not include this test class.

Comment (cataphract) [reply scenarios/appsec/test_runtime_activation.py:13]: @cbeauchesne what controls whether the condition is true or not is not the scenario being run, but rather the content of the `binaries/` directory. If there is a custom rules file there, the test will fail.

Comment (cbeauchesne) [reply scenarios/appsec/test_runtime_activation.py:13]: You're perfectly right !

**Review by cbeauchesne — APPROVED** [review]

**Comment by Julio-Guerra** [comment]
Very nice 👏 Let's put it in the feature matrix once merged, in place of the current test we've put in the 1-click enable 🙏
And please let the team know about this new test and scenario.

**Comment by Julio-Guerra** [comment]
This test should pass for nodejs (still on the feature branch) and go (merged in the main branch)
cc @simon-id @Hellzy

File: utils/proxy/rc_mocked_responses_asm_activate_only.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      "datadog/2/ASM_FEATURES/asm_features_activation/config"
+    ],
+    "roots" : [],
+    "target_files" : [
+      {
+        "path" : "datadog/2/ASM_FEATURES/asm_features_activation/config",
+        "raw" : "eyJhc20iOnsiZW5hYmxlZCI6dHJ1ZX19"
+      }
+    ],
+    "targets" : "eyJzaWduZWQiOnsiZXhwaXJlcyI6IjIwMjItMDktMTdUMTI6NDk6MTVaIiwic3BlY192ZXJzaW9uIjoiMS4wLjAiLCJ0YXJnZXRzIjp7ImRhdGFkb2cvMi9BU01fRkVBVFVSRVMvYXNtX2ZlYXR1cmVzX2FjdGl2YXRpb24vY29uZmlnIjp7ImN1c3RvbSI6eyJ2IjoxfSwiaGFzaGVzIjp7InNoYTI1NiI6IjE1OTY1OGFiODViZTcyMDc3NjFhNDExMTE3MmIwMTU1ODM5NGJmYzc0YTFmZTFkMzE0ZjIwMjNmN2M2NTZkYiJ9LCJsZW5ndGgiOjI0fX0sInZlcnNpb24iOjIzMzM3MzkzfSwic2lnbmF0dXJlcyI6W3sia2V5aWQiOiJURVNUX0tFWV9JRCIsInNpZyI6IjUwNzM1YTg5MmI5ZTViOTdhZjljMGRmMTJjY2Q5MTczNDBkYjQzNzBmYWJkNWY1M2ViNzBmNjZmM2NlNDg0NDVkMjFkMGYwZWFkZmJlZWY4NGI4MDhmYjI3NjUxYzU4NWVjZWQ4MDZkOTUzZDlmNDQzNmE2ZDQ4ZjM1MGE3ZjBjIn1dfQ=="
```
Comment (simon-id) [inline utils/proxy/rc_mocked_responses_asm_activate_only.json:13]: `targets.signed.custom.opaque_backend_state` field is missing. Nowhere was written that this field was optional so it makes the nodejs lib consider the payload as invalid and skip the update.
```suggestion
    "targets" : "eyJzaWduZWQiOnsiZXhwaXJlcyI6IjIwMjItMDktMTdUMTI6NDk6MTVaIiwic3BlY192ZXJzaW9uIjoiMS4wLjAiLCJjdXN0b20iOnsib3BhcXVlX2JhY2tlbmRfc3RhdGUiOiJhYWFhYSJ9LCJ0YXJnZXRzIjp7ImRhdGFkb2cvMi9BU01fRkVBVFVSRVMvYXNtX2ZlYXR1cmVzX2FjdGl2YXRpb24vY29uZmlnIjp7ImN1c3RvbSI6eyJ2IjoxfSwiaGFzaGVzIjp7InNoYTI1NiI6IjE1OTY1OGFiODViZTcyMDc3NjFhNDExMTE3MmIwMTU1ODM5NGJmYzc0YTFmZTFkMzE0ZjIwMjNmN2M2NTZkYiJ9LCJsZW5ndGgiOjI0fX0sInZlcnNpb24iOjIzMzM3MzkzfSwic2lnbmF0dXJlcyI6W3sia2V5aWQiOiJURVNUX0tFWV9JRCIsInNpZyI6IjUwNzM1YTg5MmI5ZTViOTdhZjljMGRmMTJjY2Q5MTczNDBkYjQzNzBmYWJkNWY1M2ViNzBmNjZmM2NlNDg0NDVkMjFkMGYwZWFkZmJlZWY4NGI4MDhmYjI3NjUxYzU4NWVjZWQ4MDZkOTUzZDlmNDQzNmE2ZDQ4ZjM1MGE3ZjBjIn1dfQ=="
```

**Review by simon-id — CHANGES_REQUESTED** [review]

**Comment by simon-id** [comment]
Also who the hell formats JSON with spaces before colons, the other json files don't

**Comment by smola** [comment]
@simon-id I applied your proposed changes, but I still see it failing in Node: https://github.com/DataDog/system-tests/actions/runs/3437494730/jobs/5732343418

**Comment by simon-id** [comment]
@smola as Julio wrote in the message just above:
>This test should pass for nodejs (still on the feature branch)

It's not expected to pass on master, hence I didn't enable the test.

File: utils/proxy/rc_mocked_responses_asm_activate_only.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    "roots" : [],
+    "target_files" : [
+      {
+        "path" : "datadog/2/ASM_FEATURES/asm_features_activation/config",
+        "raw" : "eyJhc20iOnsiZW5hYmxlZCI6dHJ1ZX19"
+      }
+    ],
+    "targets" : "eyJzaWduZWQiOnsiZXhwaXJlcyI6IjIwMjItMDktMTdUMTI6NDk6MTVaIiwic3BlY192ZXJzaW9uIjoiMS4wLjAiLCJjdXN0b20iOnsib3BhcXVlX2JhY2tlbmRfc3RhdGUiOiJhYWFhYSJ9LCJ0YXJnZXRzIjp7ImRhdGFkb2cvMi9BU01fRkVBVFVSRVMvYXNtX2ZlYXR1cmVzX2FjdGl2YXRpb24vY29uZmlnIjp7ImN1c3RvbSI6eyJ2IjoxfSwiaGFzaGVzIjp7InNoYTI1NiI6IjE1OTY1OGFiODViZTcyMDc3NjFhNDExMTE3MmIwMTU1ODM5NGJmYzc0YTFmZTFkMzE0ZjIwMjNmN2M2NTZkYiJ9LCJsZW5ndGgiOjI0fX0sInZlcnNpb24iOjIzMzM3MzkzfSwic2lnbmF0dXJlcyI6W3sia2V5aWQiOiJURVNUX0tFWV9JRCIsInNpZyI6IjUwNzM1YTg5MmI5ZTViOTdhZjljMGRmMTJjY2Q5MTczNDBkYjQzNzBmYWJkNWY1M2ViNzBmNjZmM2NlNDg0NDVkMjFkMGYwZWFkZmJlZWY4NGI4MDhmYjI3NjUxYzU4NWVjZWQ4MDZkOTUzZDlmNDQzNmE2ZDQ4ZjM1MGE3ZjBjIn1dfQ=="
+  }
+]
```
Comment (simon-id) [inline utils/proxy/rc_mocked_responses_asm_activate_only.json:15]:
```suggestion
[
  {
    "client_configs": [
      "datadog/2/ASM_FEATURES/asm_features_activation/config"
    ],
    "roots": [],
    "target_files": [
      {
        "path": "datadog/2/ASM_FEATURES/asm_features_activation/config",
        "raw": "eyJhc20iOnsiZW5hYmxlZCI6dHJ1ZX19"
      }
    ],
    "targets": "eyJzaWduZWQiOnsiZXhwaXJlcyI6IjIwMjItMDktMTdUMTI6NDk6MTVaIiwic3BlY192ZXJzaW9uIjoiMS4wLjAiLCJjdXN0b20iOnsib3BhcXVlX2JhY2tlbmRfc3RhdGUiOiJhYWFhYSJ9LCJ0YXJnZXRzIjp7ImRhdGFkb2cvMi9BU01fRkVBVFVSRVMvYXNtX2ZlYXR1cmVzX2FjdGl2YXRpb24vY29uZmlnIjp7ImN1c3RvbSI6eyJ2IjoxfSwiaGFzaGVzIjp7InNoYTI1NiI6IjE1OTY1OGFiODViZTcyMDc3NjFhNDExMTE3MmIwMTU1ODM5NGJmYzc0YTFmZTFkMzE0ZjIwMjNmN2M2NTZkYiJ9LCJsZW5ndGgiOjI0fX0sInZlcnNpb24iOjIzMzM3MzkzfSwic2lnbmF0dXJlcyI6W3sia2V5aWQiOiJURVNUX0tFWV9JRCIsInNpZyI6IjUwNzM1YTg5MmI5ZTViOTdhZjljMGRmMTJjY2Q5MTczNDBkYjQzNzBmYWJkNWY1M2ViNzBmNjZmM2NlNDg0NDVkMjFkMGYwZWFkZmJlZWY4NGI4MDhmYjI3NjUxYzU4NWVjZWQ4MDZkOTUzZDlmNDQzNmE2ZDQ4ZjM1MGE3ZjBjIn1dfQ=="
  }
]
```

**Review by simon-id — APPROVED** [review]

**Pull request merged**